src/pycram/worlds/multiverse_communication/clients.py

```diff
@@ -0,0 +1,571 @@
+import logging
+import threading
+from time import time, sleep
+
+from typing_extensions import List, Dict, Tuple, Optional
+
+from .socket import MultiverseSocket, MultiverseMetaData, SocketAddress
+from ..multiverse_datastructures.enums import MultiverseAPIName as API
+from ..multiverse_datastructures.dataclasses import RayResult, MultiverseContactPoint
+from ...datastructures.pose import Pose
+from ...datastructures.world import World
+from ...world_concepts.constraints import Constraint
+from ...world_concepts.world_object import Object, Link
+
+
+class MultiverseClient(MultiverseSocket):
+
+    def __init__(self, name: str, port: int, is_prospection_world: Optional[bool] = False, **kwargs):
+        """
+        Initialize the Multiverse client, which connects to the Multiverse server.
+        param name: The name of the client.
+        param port: The port of the client.
+        param is_prospection_world: Whether the client is connected to the prospection world.
+        """
+        meta_data = MultiverseMetaData()
+        meta_data.simulation_name = (World.prospection_world_prefix if is_prospection_world else "") + name
+        meta_data.world_name = (World.prospection_world_prefix if is_prospection_world else "") + meta_data.world_name
+        super().__init__(SocketAddress(port=str(port)), meta_data)
+        self.run()
+
+
+class MultiverseReader(MultiverseClient):
+
+    MAX_WAIT_TIME_FOR_DATA: Optional[float] = 1
+    """
+    The maximum wait time for the data in seconds.
```

Comment:
make this a datetime type and not an interpretation of int

Reply:
done